operators/src/source/gdal_source/mod.rs

```diff
@@ -420,11 +434,48 @@ impl GdalRasterLoader {
         tile_information: TileInformation,
         tile_time: TimeInterval,
     ) -> Result<RasterTile2D<T>> {
-        crate::util::spawn_blocking(move || {
-            Self::load_tile_data(&dataset_params, tile_information, tile_time)
-        })
+        let is_vsi_curl = dataset_params.file_path.starts_with("/vsicurl/");
+
+        retry(
+            dataset_params
+                .retry
+                .map(|r| r.max_retries)
+                .unwrap_or_default(),
+            GDAL_RETRY_INITIAL_BACKOFF_MS,
+            GDAL_RETRY_EXPONENTIAL_BACKOFF_FACTOR,
+            move || {
+                let ds = dataset_params.clone();
+                let file_path = ds.file_path.clone();
```

Comment:
Is cloning necessary? Maybe because it is "reused"? If not, we could remove that.

Reply:
I did not find a way to avoid cloning here. Let me know if you have any ideas.